src/main/java/com/github/susom/dbgoodies/etl/BigQueryWriter.java

```diff
@@ -272,17 +272,18 @@ private Table checkTable(BigQuery bigquery){
           break;
         case Types.DECIMAL:
         case Types.NUMERIC:
-
+          // 9 decimal digits of scale allowed by BigQuery
           BigDecimal v = r.getBigDecimalOrNull(columnNames[i]);
-          if(v!=null){
+          if(v == null){
+            row.put(columnNames[i],null);
+          }else if(v.scale()>9){
             try{
-              // 9 decimal digits of scale allowed by BigQuery
               row.put(columnNames[i],v.setScale(9, RoundingMode.HALF_UP));
             }catch (Exception e){
```

Comment:
Try not to catch general exceptions, make it more specific?

Reply:
committed the fix.